onnxruntime/test/providers/provider_test_utils.cc

```diff
@@ -733,10 +733,6 @@ void OpTester::Run(
           so.enable_mem_pattern = false;
           so.execution_mode = ExecutionMode::ORT_SEQUENTIAL;
         }
-        InferenceSession session_object{so};
-
-        for (auto& custom_session_registry : custom_session_registries_)
-          session_object.RegisterCustomRegistry(custom_session_registry);
```

Comment:
Why is this moving? Does it mean the first EP run will see different registries to subsequent ones?

Reply:
This move should not have any impact on any registers. The session_object was prematurely constructed and then destroyed when most of the providers except CPU were not present. This was impacting profiling. Since the model would not load, the session_object would complain in the destructor that it can not write a profiling information but also seemingly overwrite the one captured before.

---
In reply to: [391929448](https://github.com/microsoft/onnxruntime/pull/3197#discussion_r391929448) [](ancestors = 391929448)